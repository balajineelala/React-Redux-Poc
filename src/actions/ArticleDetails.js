import fetchArticleDetails from "../services/ArticleDetails.js";

export function getArticleDetails(param) {
  const path = param.match(/\d/g).join("");
  return async function getArticleDetailsThunk(dispatch) {
    try {
      const response = await fetchArticleDetails(path);
      dispatch(getArticleDetailsSuccess(response.data));
    } catch (error) {
      dispatch(getArticleDetailsFailure(error));
    }
  };
}

export const GET_ARTICLE_DETAILS_FAILURE = "GET_ARTICLE_DETAILS_FAILURE";

export function getArticleDetailsFailure(error) {
  return {
    type: GET_ARTICLE_DETAILS_FAILURE,
    error,
  };
}

export const GET_ARTICLE_DETAILS_SUCCESS = "GET_ARTICLE_DETAILS_SUCCESS";

export function getArticleDetailsSuccess(articleDetails) {
  return {
    type: GET_ARTICLE_DETAILS_SUCCESS,
    articleDetails,
  };
}
