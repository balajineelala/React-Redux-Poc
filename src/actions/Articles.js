import fetchArticles from "../services/Articles.js";

export function getArticles() {
  return async function getArticlesThunk(dispatch) {
    try {
      const response = await fetchArticles();
      dispatch(getArticlesSuccess(response.data));
    } catch (error) {
      dispatch(getArticlesFailure(error));
    }
  };
}

export const GET_ARTICLES_FAILURE = "GET_ARTICLES_FAILURE";

export function getArticlesFailure(error) {
  return {
    type: GET_ARTICLES_FAILURE,
    error,
  };
}

export const GET_ARTICLES_SUCCESS = "GET_ARTICLES_SUCCESS";

export function getArticlesSuccess(articlesList) {
  return {
    type: GET_ARTICLES_SUCCESS,
    articlesList,
  };
}
