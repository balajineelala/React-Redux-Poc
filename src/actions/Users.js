import fetchUsers from "../services/Users.js";

export function getUsers() {
  return async function getUsersThunk(dispatch) {
    try {
      const response = await fetchUsers();
      dispatch(getUsersSuccess(response.data));
    } catch (error) {
      dispatch(getUsersFailure(error));
    }
  };
}

export const GET_USERS_FAILURE = "GET_USERS_FAILURE";

export function getUsersFailure(error) {
  return {
    type: GET_USERS_FAILURE,
    error,
  };
}

export const GET_USERS_SUCCESS = "GET_USERS_SUCCESS";

export function getUsersSuccess(usersList) {
  return {
    type: GET_USERS_SUCCESS,
    usersList,
  };
}
