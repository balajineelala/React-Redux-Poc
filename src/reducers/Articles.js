import {
  GET_ARTICLES_SUCCESS,
  GET_ARTICLES_FAILURE,
} from "../actions/Articles.js";

const initialState = {
  articlesList: [],
};

function Articles(state = initialState, action) {
  switch (action.type) {
    case GET_ARTICLES_SUCCESS:
      return {
        ...state,
        articlesList: action.articlesList,
      };
    case GET_ARTICLES_FAILURE:
      return {
        error: action.error,
      };
    default:
      return state;
  }
}

export default Articles;
