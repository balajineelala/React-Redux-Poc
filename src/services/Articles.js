import axios from "axios";

export default function fetchArticles() {
  return axios.get("https://jsonplaceholder.typicode.com/posts");
}
