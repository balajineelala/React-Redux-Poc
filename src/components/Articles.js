import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { getArticles } from "../actions/Articles";
import ArticlesList from "./ArticlesList";

function Articles() {
  const { articlesList } = useSelector((state) => state.articles);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getArticles());
  }, [dispatch]);

  return (
    <div>
      <ArticlesList articles={articlesList} />
    </div>
  );
}

export default React.memo(Articles);
