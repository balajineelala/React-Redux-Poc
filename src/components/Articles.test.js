import React from "react";
import Articles from "./Articles";
import promise from "redux-promise-middleware";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { getArticles } from "../actions/Articles";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"), // use actual for all non-hook parts
  useParams: () => ({
    articles: "articles",
  }),
  useRouteMatch: () => ({ url: "/articles" }),
}));

const mockStore = configureMockStore([thunk, promise]);

jest.mock("axios");

const initialState = { articles: { articlesList: [] } };
let store, wrapper;

describe("Articles hook", () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Articles />
      </Provider>
    );
  });
  //Snap Shot Testing for Hook Components
  it("Articles snapshot testing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
