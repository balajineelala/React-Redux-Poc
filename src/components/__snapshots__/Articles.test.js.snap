// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles hook Articles snapshot testing 1`] = `ReactWrapper {}`;
